src/main/webapp/app/shared/util/firebase/firebase-reviewable-crud-store.spec.ts

```diff
@@ -0,0 +1,160 @@
+const mockUpdate = jest.fn().mockImplementation((db, value) => Promise.resolve());
+const mockRemove = jest.fn().mockImplementation((db, value) => {});
+const mockRef = jest.fn().mockImplementation(db => {});
+
+import 'jest-expect-message';
+import { FirebaseReviewableCrudStore } from './firebase-reviewable-crud-store';
+import { FirebaseMetaStore } from 'app/stores/firebase/firebase.meta.store';
+import FirebaseStore from 'app/stores/firebase/firebase.store';
+
+jest.mock('firebase/database', () => {
+  return {
+    ref: mockRef,
+    update: mockUpdate,
+    remove: mockRemove,
+  };
+});
+
+const updateGeneMetaContentMock = jest.spyOn(FirebaseMetaStore.prototype, 'updateGeneMetaContent').mockReturnValue(Promise.resolve());
+const updateGeneReviewUuidMock = jest.spyOn(FirebaseMetaStore.prototype, 'updateGeneReviewUuid').mockReturnValue(Promise.resolve());
+
+describe('FirebaseReviewableCrudStore', () => {
+  class TestRootStore {
+    firebaseStore;
+    firebaseMetaStore;
+    authStore;
+    constructor() {
+      this.firebaseStore = new FirebaseStore(this as any);
+      this.firebaseMetaStore = new FirebaseMetaStore(this as any);
+      this.authStore = { fullName: 'test user' };
+    }
+  }
+
+  let rootStore = undefined;
+  const defaultDate = new Date('2023-01-01');
+
+  const reset = () => {
+    jest.useFakeTimers().setSystemTime(defaultDate);
+    rootStore = new TestRootStore();
+    jest.clearAllMocks();
+  };
+
+  describe('updateReviewableContent', () => {
+    beforeEach(() => reset());
+
+    it('should correctly update field that was already reviewed', async () => {
+      const store = new FirebaseReviewableCrudStore<any>(rootStore);
+      store.data = { key: 'original', key_uuid: 'fake-uuid' };
+
+      await store.updateReviewableContent('Gene/ABL1', 'key', 'new value');
+
+      expect(mockUpdate).toHaveBeenCalled();
+
+      // Check if field and field_review were updated
+      // lastReviewed should be set because the field was already reviewed and we are making a new change
+      const updateValueArg = mockUpdate.mock.calls[0][1];
+      expect(updateValueArg).toBeDefined();
+      expect(updateValueArg).toEqual({
+        key: 'new value',
```

Comment:
I think the reason why I didn't check the `uuid` is because when we create the object (let's say Gene), the `_uuid` field should already be generated.

I can add an extra check to make sure `_uuid` field is present. If not, then we set it.

Reply:
Good point. Fine to not test here, but we should add a test to make sure all the properties in a newly created model have `_uuid` created. We can start with Gene, then expand to Mutation etc.